packages/go/analysis/ad/ad_test.go

```diff
@@ -0,0 +1,339 @@
+
+	var fetchedNode *graph.Node
+	err := graphDB.ReadTransaction(ctx, func(tx graph.Transaction) error {
+		filteredNode, err := tx.Nodes().Filterf(func() graph.Criteria {
+			criterias := make([]graph.Criteria, len(queryCriterias))
+			for i := range queryCriterias {
+				criterias[i] = queryCriterias[i]
+			}
+			return query.And(criterias...)
+		}).First()
+		if err != nil {
+			return err
+		}
+		fetchedNode = filteredNode
+
+		return nil
+	})
+	return fetchedNode, err
+}
+
+func assertNodeExists(
+	t *testing.T,
+	ctx context.Context,
+	graphDB graph.Database,
+	expectedNode *graph.Node,
+	queryCriterias ...*cypher.Comparison,
+) {
+	t.Helper()
+
+	domainSID, err := expectedNode.Properties.Get(ad.DomainSID.String()).String()
+	require.NoError(t, err)
+	domainFQDN, err := expectedNode.Properties.Get(ad.DomainFQDN.String()).String()
+	require.NoError(t, err)
+	nodeName, err := expectedNode.Properties.Get(common.Name.String()).String()
+	require.NoError(t, err)
+
+	defaultQueryCriterias := []*cypher.Comparison{
+		query.Equals(query.NodeProperty(ad.DomainSID.String()), domainSID),
+		query.Equals(query.NodeProperty(ad.DomainFQDN.String()), domainFQDN),
+		query.Equals(query.NodeProperty(common.Name.String()), nodeName),
+	}
+
+	queryCriterias = append(queryCriterias, defaultQueryCriterias...)
+	fetchedNode, err := fetchNode(t, ctx, graphDB, queryCriterias...)
+	require.NoError(t, err)
+	require.NotNil(t, fetchedNode)
+}
+
+func createNode(
+	t *testing.T,
+	ctx context.Context,
+	graphDB graph.Database,
+	nodeToCreate *graph.Node,
+) *graph.Node {
+	t.Helper()
+	require.NotNil(t, graphDB)
+	require.NotNil(t, nodeToCreate)
+
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	var createdNode *graph.Node
+	var err error
+	err = graphDB.WriteTransaction(ctx, func(tx graph.Transaction) error {
+		createdNode, err = tx.CreateNode(nodeToCreate.Properties, nodeToCreate.Kinds...)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	require.NoError(t, err)
+	require.NotNil(t, createdNode)
+
+	assertNodeExists(t, ctx, graphDB, createdNode)
+	return createdNode
+}
+
+func createCollectedDomainNode(
+	t *testing.T,
+	ctx context.Context,
+	graphDB graph.Database,
+) *graph.Node {
+	return createNode(
+		t,
+		ctx,
+		graphDB,
+		generateCollectedDomain(),
+	)
+}
+
+func createWellKnownGroup(
+	t *testing.T,
+	ctx context.Context,
+	graphDB graph.Database,
+	domainNode *graph.Node,
+	sidSuffix wellknown.SIDSuffix,
+	nodeNamePrefix wellknown.NodeNamePrefix,
+) *graph.Node {
+	return createNode(
+		t,
+		ctx,
+		graphDB,
+		generateWellKnownGroup(
+			t,
+			domainNode,
+			sidSuffix,
+			nodeNamePrefix,
+		),
+	)
+}
+
+func generateCollectedDomain() *graph.Node {
+	return &graph.Node{
+		Kinds: graph.Kinds{
+			ad.Domain,
+		},
+		Properties: graph.AsProperties(graph.PropertyMap{
+			common.Collected: true,
+			common.Name:      faker.DomainName(),
+			ad.DomainSID:     xid.New().String(),
+			ad.DomainFQDN:    faker.DomainName(),
+		}),
+	}
+}
```

Comment:
I'd recommend at a minimum removing the faker/xid calls and using constants here. Ideally, I think the call stack for generating the graph is a little too tall and should be flattened a bit more (i.e.  don't use a function to create the node, just format it inline upstream).

After further discussions, I think it would be wise to drop the faker and xid deps, as they are not providing sufficient value here and we don't currently have a need for them in other testing initiatives right now. It's not that they're bad dependencies, it's just that we want to avoid unnecessary dependencies and the bar for usefulness needs to be higher